### PR TITLE
P5 inno 166 Project Showcase Page Creation

### PR DESCRIPTION
**Pull request by nikitaszvan**
For this PR:

Completed:

- [x] Created logic for fetching project submission details (i.e., submitted project details, comments, judges comments, etc) from firebase db.

- [x] Created logic from fetching media (i.e, images) from firebase db and bootstrapped shadcn carousel to display media.

- [x] Created functional comment section that users can leave on the completed hackathon project and it will save to the project submission collection.

- [x] Created routing to project showcase page

Will Improve:

- [ ] Will continue styling, even more so once the designs are completed 

Need Help:

- [ ] General feedback on code for the best practices / edge cases

### CODE REVIEW

File: src/pages/ProjectShowcasePage/ProjectShowcasePage.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                try {
+                    const urls = await Promise.all(
+                        submission.imageFiles.map(async (file: string) => {
+                            const imageRef = ref(storage, file);
+                            return await getDownloadURL(imageRef);
+                        })
+                    );
+                    setImageUrls(urls);
+                } catch (error) {
+                    console.error(error);
```
Comment (strawberrie68) [inline src/pages/ProjectShowcasePage/ProjectShowcasePage.tsx]: Consider writing more descriptive error message ie console.error('Failed to fetch images:', error);

File: src/pages/ProjectShowcasePage/ProjectShowcasePage.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    CarouselContent,
+    CarouselItem,
+    CarouselNext,
+    CarouselPrevious
+} from '../../components/ui/carousel';
+import CommunityCommentSection from '../../components/CommunityCommentSection/CommunityCommentSection';
+import JudgesCommentSection from '../../components/JudgesCommentSection/JudgesCommentSection';
+
+const ProjectShowcasePage = () => {
+    const { submissionId } = useParams();
```
Comment (strawberrie68) [inline src/pages/ProjectShowcasePage/ProjectShowcasePage.tsx]: const { submissionId } = useParams<{ submissionId: string }>();

File: src/pages/ProjectShowcasePage/ProjectShowcasePage.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    } = submission || {};
+
+    return (
+        <div className='w-full h-full flex flex-col items-center font-poppins'>
+            <h1>{event ? event.title : 'Event not found'} submission # {submissionId} by {teamName}</h1>
+            <Carousel className='w-[20%] h-[20%]'>
+                <CarouselContent>
+                    {imageUrls.map((imageUrl, index) => (
+                        <CarouselItem key={index} className='w-[50%] h-[50%]'>
+                            <img src={imageUrl} alt={`Image ${index}`} />
```
Comment (strawberrie68) [inline src/pages/ProjectShowcasePage/ProjectShowcasePage.tsx]: Consider a more descriptive alt tag ie Submission image ${index + 1}`

File: src/pages/ProjectShowcasePage/ProjectShowcasePage.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+                        </CarouselItem>
+                    ))}
+                </CarouselContent>
+                <CarouselPrevious />
+                <CarouselNext />
+            </Carousel>
+            <p>Design Features: {designFeatures}</p>
```
Comment (strawberrie68) [inline src/pages/ProjectShowcasePage/ProjectShowcasePage.tsx]: Consider more semantic HTML

File: src/pages/ProjectShowcasePage/ProjectShowcasePage.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+            <h1>{event ? event.title : 'Event not found'} submission # {submissionId} by {teamName}</h1>
+            <Carousel className='w-[20%] h-[20%]'>
+                <CarouselContent>
+                    {imageUrls.map((imageUrl, index) => (
+                        <CarouselItem key={index} className='w-[50%] h-[50%]'>
+                            <img src={imageUrl} alt={`Image ${index}`} />
+                        </CarouselItem>
+                    ))}
+                </CarouselContent>
+                <CarouselPrevious />
```
Comment (strawberrie68) [inline src/pages/ProjectShowcasePage/ProjectShowcasePage.tsx]: Consider adding aria-label like "Previous image" or "Next image"

File: src/pages/ProjectShowcasePage/ProjectShowcasePage.tsx (hunk 1 above)
Comment (nikitaszvan) [reply src/pages/ProjectShowcasePage/ProjectShowcasePage.tsx]: Resolved

File: src/pages/ProjectShowcasePage/ProjectShowcasePage.tsx (hunk 2 above)
Comment (nikitaszvan) [reply src/pages/ProjectShowcasePage/ProjectShowcasePage.tsx]: Resolved

File: src/pages/ProjectShowcasePage/ProjectShowcasePage.tsx (hunk 3 above)
Comment (nikitaszvan) [reply src/pages/ProjectShowcasePage/ProjectShowcasePage.tsx]: Resolved

File: src/pages/ProjectShowcasePage/ProjectShowcasePage.tsx (hunk 4 above)
Comment (nikitaszvan) [reply src/pages/ProjectShowcasePage/ProjectShowcasePage.tsx]: I made an attempt to code this section in a more semantic HTML, but will continue to improve during the styling

File: src/pages/ProjectShowcasePage/ProjectShowcasePage.tsx (hunk 5 above)
Comment (nikitaszvan) [reply src/pages/ProjectShowcasePage/ProjectShowcasePage.tsx]: Resolved

**Comment by nikitaszvan** [comment]
@strawberrie68 Thanks Michelle, your feedback in this review has been very helpful. I have applied all the suggested changes.

**Pull request merged**